spec.md

```diff
@@ -388,13 +388,34 @@ both appear within the same JSON object.
   or application protocol's content-type metadata property. Normative rules
   for the binary mode and the content-type metadata mapping can be found 
   in the respective transport mapping specifications.
-    
+  
 * Constraints:
   * OPTIONAL
   * If present, MUST adhere to the format specified in
     [RFC 2046](https://tools.ietf.org/html/rfc2046)
 * For Media Type examples see [IANA Media Types](http://www.iana.org/assignments/media-types/media-types.xhtml)
 
+### application-properties
```

Comment:
If this is intended to be a general purpose bucket for custom key/value pairs that are not official spec attributes or extension attributes, I actually think "properties" was a better name. The name "application-properties" implies that the properties *describe* the application, but most of the examples that have been discussed refer to properties that describe the source of the event and/or the event itself.

That said, I'm personally in favor of allowing arbitrary properties as top-level attributes since the extensions have been elevated and therefore serialization implementations etc will already need to deal with the classification of top-level keys (core, extensions, or custom).